### DISCUSSION TITLE
How to use `Prisma.join` with the feature `improvedQueryRaw`?

### DISCUSSION THREAD

**Discussion by wongsean** [prisma/prisma#13872] [Q&A] [answered]
Hello! I upgraded Prisma to "3.15.1" and turned on the "improvedQueryRaw" feature. 
I have a problem when using `Prisma.join` on a UUID field. Either `WHERE "id" IN (${Prisma.join(itemIds)}::uuid)` or `WHERE "id" IN (${Prisma.join(itemIds})::uuid[]` doesn't work. (ERROR: operator does not exist: uuid = text)

```typescript
prisma.$queryRaw`
  SELECT * FROM "item"
  WHERE "id" in (${Prisma.join(itemIds)})`

model Item {
  id    String.   @id @default(uuid()) @db.Uuid
}
```

I am wondering is there an approach that I can cast multiple texts to the correct type?

**Comment by matthewmueller** [comment]
Hey @wongsean can you [log the underlying SQL query](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging) and share it with us? That would help us understand what's happening a bit better.

**Reply by wongsean** [reply]
Hi @matthewmueller, thank you for your help! 

Commonly the SQL query would be:
`SELECT * FROM "item" WHERE "id" IN ($1, $2, $3)` with params like `['a9467443-2e05-4609-8eea-8d7081d97ca6', ...]`
However, I believe it's not permitted with the feature "improvedQueryRaw" enabled.

Using `WHERE "id" IN (${Prisma.join(itemIds)}::uuid)`, it simply casts the last parameter, which, as evidence, works fine when there is only one parameter. The query log would be: `SELECT * FROM "item" WHERE "id" IN ($1, $2, $3::uuid)`.

I also tried `WHERE "id" IN (${Prisma.join(itemIds})::uuid[]`, the query log is like `SELECT * FROM "item" WHERE "id" IN ($1, $2, $3)::uuid[]`. I guess it's not a valid type cast since same error still retains.

**Reply by wongsean** [reply]
Hi @matthewmueller.
With releasing prisma@4, the feature "improvedQueryRaw" is active by default. I wonder if there is a way to do type-casts for multi-params in raw queries now?

**Comment by oleksandr-korolchuk-moc** [comment, marked as answer]
Hi @wongsean @matthewmueller . I have the same issue. Do you have an answer on how to fix it? Thanks in advance.

**Reply by wongsean** [reply]
issue https://github.com/prisma/prisma/issues/14956

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
